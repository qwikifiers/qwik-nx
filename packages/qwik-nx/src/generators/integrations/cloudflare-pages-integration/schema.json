{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "CloudflarePagesIntegration",
  "title": "",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "Project for the integration to be added",
      "$default": {
        "$source": "argv",
        "index": 0
      }
    },
    "skipFormat": {
      "description": "Skip formatting files.",
      "type": "boolean",
      "default": false
    }
  },
  "required": ["project"]
}
