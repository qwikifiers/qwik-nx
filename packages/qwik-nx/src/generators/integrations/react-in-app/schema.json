{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "ReactInApp",
  "title": "React In App integration",
  "description": "Add qwikified React components to your Qwik app",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "Name of the project to add React integration to"
    },
    "installMUIExample": {
      "type": "boolean",
      "description": "Add dependencies for Material UI and qwikified example component, that uses it",
      "x-priority": "important",
      "default": true
    },
    "skipFormat": {
      "description": "Skip formatting files.",
      "type": "boolean",
      "x-priority": "internal",
      "default": false
    }
  },
  "required": ["project"]
}
