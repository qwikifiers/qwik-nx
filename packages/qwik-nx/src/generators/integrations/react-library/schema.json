{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "ReactLibrary",
  "title": "React Library",
  "description": "Add qwikified React components as a separate library",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "Library name.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the library?",
      "pattern": "^[a-zA-Z].*$"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the lib is placed."
    },
    "installMUIExample": {
      "type": "boolean",
      "description": "Add dependencies for Material UI and qwikified example component, that uses it",
      "x-priority": "important",
      "default": true
    },
    "targetApps": {
      "oneOf": [
        { "type": "string" },
        { "type": "array", "items": { "type": "string", "uniqueItems": true } }
      ],
      "description": "The names of the applications that should be configured to work with react components.",
      "x-prompt": "What apps are you going to use this integration with? (Comma delimited, leave empty for none)",
      "x-priority": "important"
    },
    "buildable": {
      "type": "boolean",
      "default": false,
      "description": "Generate a buildable library.",
      "x-priority": "important"
    },
    "linter": {
      "description": "The tool to use for running lint checks.",
      "type": "string",
      "enum": ["eslint", "none"],
      "default": "eslint"
    },
    "unitTestRunner": {
      "type": "string",
      "enum": ["vitest", "none"],
      "description": "Test runner to use for unit tests.",
      "default": "vitest"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the library (used for linting)."
    },
    "skipFormat": {
      "description": "Skip formatting files.",
      "type": "boolean",
      "x-priority": "internal",
      "default": false
    },
    "importPath": {
      "type": "string",
      "description": "The library name used to import it, like `@myorg/my-awesome-lib`."
    },
    "strict": {
      "type": "boolean",
      "description": "Whether to enable `tsconfig` strict mode or not.",
      "default": true
    },
    "storybookConfiguration": {
      "description": "Whether to include storybook configuration for the generated library.",
      "type": "boolean",
      "default": false
    }
  },
  "required": ["name"]
}
