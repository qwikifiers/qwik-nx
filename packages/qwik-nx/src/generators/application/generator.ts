import {
  addProjectConfiguration,
  formatFiles,
  generateFiles,
  GeneratorCallback,
  joinPathFragments,
  names,
  readProjectConfiguration,
  Tree,
  updateProjectConfiguration,
} from '@nrwl/devkit';
import { NormalizedSchema, normalizeOptions } from './utils/normalize-options';
import { viteConfigurationGenerator } from '@nrwl/vite';
import { QwikAppGeneratorSchema } from './schema';
import { Linter } from '@nrwl/linter';
import { addStyledModuleDependencies } from '../../utils/add-styled-dependencies';
import { runTasksInSerial } from '@nrwl/workspace/src/utilities/run-tasks-in-serial';
import { configureEslint } from '../../utils/configure-eslint';
import { addCommonQwikDependencies } from '../../utils/add-common-qwik-dependencies';

function addFiles(tree: Tree, options: NormalizedSchema) {
  const templateOptions = {
    ...options,
    ...names(options.name),
    template: '',
  };
  generateFiles(
    tree,
    joinPathFragments(__dirname, 'files'),
    options.appProjectRoot,
    templateOptions
  );
}

async function configureVite(
  tree: Tree,
  options: NormalizedSchema
): Promise<GeneratorCallback> {
  const tsConfigPath = joinPathFragments(
    options.appProjectRoot,
    'tsconfig.json'
  );
  const tsConfig = tree.read(tsConfigPath);
  const includeVitest = options.unitTestRunner === 'vitest';

  const callback = await viteConfigurationGenerator(tree, {
    project: options.projectName,
    newProject: false,
    includeVitest,
    includeLib: false,
    uiFramework: 'none',
    inSourceTests: false,
  });
  tree.delete(joinPathFragments(options.appProjectRoot, 'index.html'));

  // overwrite tsconfig back after "viteConfigurationGenerator"
  tree.write(tsConfigPath, tsConfig);

  if (includeVitest) {
    const projectConfig = readProjectConfiguration(tree, options.projectName);
    const testTarget = projectConfig.targets['test'];
    testTarget.outputs = ['{workspaceRoot}/coverage/{projectRoot}'];
    updateProjectConfiguration(tree, options.projectName, projectConfig);
  }

  return callback;
}

export default async function (tree: Tree, options: QwikAppGeneratorSchema) {
  const normalizedOptions = normalizeOptions(tree, options);
  const tasks: GeneratorCallback[] = [];

  addProjectConfiguration(tree, normalizedOptions.projectName, {
    root: normalizedOptions.appProjectRoot,
    name: normalizedOptions.projectName,
    projectType: 'application',
    sourceRoot: `${normalizedOptions.appProjectRoot}/src`,
    // TODO: revisit later, now "viteConfigurationGenerator" will fail if there's no build target specified
    targets: { build: { options: {} } },
    tags: normalizedOptions.parsedTags,
  });

  addFiles(tree, normalizedOptions);

  tasks.push(await configureVite(tree, normalizedOptions));

  if (normalizedOptions.linter === Linter.EsLint) {
    tasks.push(configureEslint(tree, normalizedOptions.projectName, true));
  }

  if (normalizedOptions.style !== 'none') {
    tasks.push(
      addStyledModuleDependencies(tree, normalizedOptions.styleExtension)
    );
  }

  tasks.push(addCommonQwikDependencies(tree));

  if (!options.skipFormat) {
    await formatFiles(tree);
  }

  return runTasksInSerial(...tasks);
}
