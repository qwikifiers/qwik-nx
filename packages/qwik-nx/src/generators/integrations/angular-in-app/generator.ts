import { Tree } from '@nx/devkit';
import { AngularInAppGeneratorSchema } from './schema';
import { ensureNxDependencies } from './../../../utils/ensure-nx-dependencies';

export async function angularInAppGenerator(
  tree: Tree,
  schema: AngularInAppGeneratorSchema
) {
  ensureNxDependencies();

  const { angularInAppGeneratorInternal } = await import(
    './generator-internal'
  );
  return angularInAppGeneratorInternal(tree, schema);
}

export default angularInAppGenerator;
