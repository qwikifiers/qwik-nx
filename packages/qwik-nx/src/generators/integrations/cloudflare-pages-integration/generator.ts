import {
  addDependenciesToPackageJson,
  formatFiles,
  generateFiles,
  GeneratorCallback,
  joinPathFragments,
  names,
  offsetFromRoot,
  ProjectConfiguration,
  readProjectConfiguration,
  TargetConfiguration,
  Tree,
  updateProjectConfiguration,
} from '@nrwl/devkit';
import {
  getIntegrationConfigurationName,
  IntegrationName,
} from '../../../utils/integration-configuration-name';
import { nxCloudflareWrangler, wranglerVersion } from '../../../utils/versions';
import { CloudflarePagesIntegrationGeneratorSchema } from './schema';

interface NormalizedOptions {
  offsetFromRoot: string;
  projectConfig: ProjectConfiguration;
}

export async function cloudflarePagesIntegrationGenerator(
  tree: Tree,
  options: CloudflarePagesIntegrationGeneratorSchema
) {
  const config = readProjectConfiguration(tree, options.project);
  if (config.projectType !== 'application') {
    throw new Error(
      'Cannot setup cloudflare integration for the given project.'
    );
  }
  if (config.targets?.['build']?.executor !== 'qwik-nx:build') {
    throw new Error(
      'Project contains invalid configuration. ' +
        'If you encounter this error within a Qwik project, make sure you have run necessary Nx migrations for qwik-nx plugin.'
    );
  }

  const configurationName = getIntegrationConfigurationName(
    IntegrationName.Cloudflare,
    config
  );
  const deployTargetName = config.targets['deploy']
    ? 'deploy.cloudflare'
    : 'deploy';

  const normalizedOptions = normalizeOptions(config);
  (config.targets['build']?.configurations ?? {})[configurationName] = {};
  (config.targets['build.ssr'].configurations ??= {})[configurationName] =
    getBuildSSRTargetCloudflareConfiguration(normalizedOptions);
  config.targets[deployTargetName] = getDeployTarget(normalizedOptions);
  config.targets['preview-cloudflare'] =
    getCloudflarePreviewTarget(normalizedOptions);
  config.targets['build-cloudflare'] = getIntermediateDependsOnTarget(
    normalizedOptions,
    configurationName
  );

  updateProjectConfiguration(tree, options.project, config);

  addFiles(tree, normalizedOptions);

  if (!options.skipFormat) {
    await formatFiles(tree);
  }

  return addCloudflarePagesDependencies(tree);
}

function getBuildSSRTargetCloudflareConfiguration(options: NormalizedOptions) {
  return {
    configFile: `${options.projectConfig.root}/adapters/cloudflare-pages/vite.config.ts`,
  };
}

function getDeployTarget(options: NormalizedOptions): TargetConfiguration {
  return {
    executor: '@k11r/nx-cloudflare-wrangler:deploy-page',
    options: {
      dist: `dist/${options.projectConfig.root}/client`,
    },
    dependsOn: ['build-cloudflare'],
  };
}

function getCloudflarePreviewTarget(
  options: NormalizedOptions
): TargetConfiguration {
  return {
    executor: '@k11r/nx-cloudflare-wrangler:serve-page',
    options: {
      dist: `dist/${options.projectConfig.root}/client`,
    },
    dependsOn: ['build-cloudflare'],
  };
}

/** Currently it's not possible to depend on a target with a specific configuration, that's why intermediate one is required */
function getIntermediateDependsOnTarget(
  options: NormalizedOptions,
  configurationName: string
): TargetConfiguration {
  return {
    executor: 'nx:run-commands',
    options: {
      command: `npx nx run ${options.projectConfig.name}:build:${configurationName}`,
    },
  };
}

function normalizeOptions(
  projectConfig: ProjectConfiguration
): NormalizedOptions {
  return {
    projectConfig,
    offsetFromRoot: offsetFromRoot(projectConfig.root),
  };
}

function addFiles(tree: Tree, options: NormalizedOptions): void {
  const templateOptions = {
    ...names(options.projectConfig.name!),
    projectRoot: options.projectConfig.root,
    offsetFromRoot: options.offsetFromRoot,
  };
  generateFiles(
    tree,
    joinPathFragments(__dirname, 'files'),
    options.projectConfig.root,
    templateOptions
  );
}

function addCloudflarePagesDependencies(tree: Tree): GeneratorCallback {
  return addDependenciesToPackageJson(
    tree,
    {},
    {
      wrangler: wranglerVersion,
      '@k11r/nx-cloudflare-wrangler': nxCloudflareWrangler,
    }
  );
}

export default cloudflarePagesIntegrationGenerator;
