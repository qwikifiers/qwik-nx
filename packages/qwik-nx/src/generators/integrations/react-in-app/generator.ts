import {
  formatFiles,
  generateFiles,
  joinPathFragments,
  ProjectType,
  readProjectConfiguration,
  Tree,
} from '@nx/devkit';
import * as path from 'path';
import { ReactInAppGeneratorSchema } from './schema';
import { reactInit } from '../../../utils/react/init';

interface NormalizedSchema extends ReactInAppGeneratorSchema {
  sourceRoot: string;
  projectRoot: string;
  projectType: ProjectType;
}

function normalizeOptions(
  tree: Tree,
  options: ReactInAppGeneratorSchema
): NormalizedSchema {
  const projectConfig = readProjectConfiguration(tree, options.project);

  return {
    ...options,
    installMUIExample: options.installMUIExample !== false,
    sourceRoot: projectConfig.sourceRoot ?? projectConfig.root + '/src',
    projectRoot: projectConfig.root,
    projectType: projectConfig.projectType!,
  };
}

function addFiles(tree: Tree, normalizedOptions: NormalizedSchema): void {
  const filePath = normalizedOptions.installMUIExample ? 'mui' : 'demo';
  generateFiles(
    tree,
    path.join(__dirname, 'files', filePath),
    joinPathFragments(normalizedOptions.sourceRoot, 'routes/react'),
    {}
  );
}

export async function reactInAppGenerator(
  tree: Tree,
  schema: ReactInAppGeneratorSchema
) {
  const normalizedOptions = normalizeOptions(tree, schema);

  if (normalizedOptions.projectType !== 'application') {
    throw new Error(
      `Only applications are supported, "${normalizedOptions.project}" is a library.`
    );
  }

  const demoFilePath = joinPathFragments(
    normalizedOptions.sourceRoot,
    'integrations/react'
  );

  if (tree.exists(demoFilePath)) {
    throw new Error(
      `Looks like react integration has already been configured for ${normalizedOptions.project}. "${demoFilePath}" already exists.`
    );
  }

  const initCallback = reactInit(tree, {
    demoFilePath: joinPathFragments(
      normalizedOptions.sourceRoot,
      'integrations/react'
    ),
    installMUIExample: !!normalizedOptions.installMUIExample,
    projectRoot: normalizedOptions.projectRoot,
  });
  addFiles(tree, normalizedOptions);
  if (!normalizedOptions.skipFormat) {
    await formatFiles(tree);
  }

  return initCallback;
}

export default reactInAppGenerator;
