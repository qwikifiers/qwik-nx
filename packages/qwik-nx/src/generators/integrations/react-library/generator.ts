import {
  formatFiles,
  generateFiles,
  GeneratorCallback,
  joinPathFragments,
  output,
  readProjectConfiguration,
  Tree,
} from '@nrwl/devkit';
import * as path from 'path';
import { ReactLibraryGeneratorSchema } from './schema';
import { normalizeOptions as libraryNormalizeOptions } from './../../library/utils/normalize-options';
import libraryGenerator from '../../library/generator';
import {
  addReactPluginToViteConfig,
  reactInit,
} from '../../../utils/react/init';
import { runTasksInSerial } from '@nrwl/workspace/src/utilities/run-tasks-in-serial';

interface NormalizedSchema extends ReactLibraryGeneratorSchema {
  installMUIExample: boolean;
  targetApps: string[];
  projectName: string;
  projectRoot: string;
  projectDirectory: string;
  parsedTags: string[];
}

function normalizeOptions(
  tree: Tree,
  options: ReactLibraryGeneratorSchema
): NormalizedSchema {
  const normalizedLibraryOptions = libraryNormalizeOptions(tree, {
    ...options,
    style: 'none',
  });

  let targetApps: string[] = [];
  if (Array.isArray(options.targetApps)) {
    targetApps = options.targetApps.filter(Boolean);
  } else if (
    typeof options.targetApps === 'string' &&
    options.targetApps.trim()
  ) {
    // can be of type "string" if value comes from the x-prompt
    targetApps = options.targetApps
      .split(',')
      .map((v) => v.trim())
      .filter(Boolean);
  }

  return {
    ...normalizedLibraryOptions,
    installMUIExample: options.installMUIExample !== false,
    targetApps,
  };
}

function addFiles(tree: Tree, options: NormalizedSchema) {
  generateFiles(
    tree,
    path.join(__dirname, 'files'),
    options.projectRoot,
    options
  );
}

function updateTargetApps(tree: Tree, targetApps: string[]) {
  if (!targetApps?.length) {
    output.warn({
      title: `You haven't provided any "targetApps" to be configured.`,
      bodyLines: [
        'Qwik React integration requires "qwikReact()" plugin to be included for all apps where qwikified react components are used.',
        'You can still import the plugin manually.',
      ],
    });
  } else {
    for (const targetApp of new Set(targetApps)) {
      const config = readProjectConfiguration(tree, targetApp);
      if (config.projectType !== 'application') {
        throw new Error(`"${targetApp}" project is not an application`);
      }
      addReactPluginToViteConfig(tree, config.root);
    }
  }
}

export async function reactLibraryGenerator(
  tree: Tree,
  schema: ReactLibraryGeneratorSchema
) {
  const normalizedOptions = normalizeOptions(tree, schema);
  const tasks: GeneratorCallback[] = [];

  const installTask = await libraryGenerator(tree, {
    ...normalizedOptions,
    skipFormat: true,
    generateComponent: false,
  });
  tasks.push(installTask);

  const initTask = reactInit(tree, {
    demoFilePath: joinPathFragments(
      normalizedOptions.projectRoot,
      'src/lib/integration'
    ),
    installMUIExample: !!normalizedOptions.installMUIExample,
    projectRoot: normalizedOptions.projectRoot,
  });
  tasks.push(initTask);

  addFiles(tree, normalizedOptions);

  updateTargetApps(tree, normalizedOptions.targetApps);

  if (!normalizedOptions.skipFormat) {
    await formatFiles(tree);
  }

  return runTasksInSerial(...tasks);
}

export default reactLibraryGenerator;
