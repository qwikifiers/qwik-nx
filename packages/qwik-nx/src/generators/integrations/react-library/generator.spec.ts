import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';
import { Tree, readProjectConfiguration } from '@nrwl/devkit';

import { reactLibraryGenerator } from './generator';
import { ReactLibraryGeneratorSchema } from './schema';
import { getFormattedListChanges } from '../../../utils/testing-generators';
import appGenerator from '../../application/generator';

describe('integrations/react-library generator', () => {
  let appTree: Tree;
  const options: ReactLibraryGeneratorSchema = { name: 'mylib' };

  beforeEach(() => {
    appTree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
  });

  it('should run successfully', async () => {
    await reactLibraryGenerator(appTree, options);
    const config = readProjectConfiguration(appTree, 'mylib');

    expect(config.targets!['build']).not.toBeDefined();
    expect(config.targets!['test']).toBeDefined();
    expect(config.targets!['lint']).toBeDefined();

    expect(config).toMatchSnapshot();
    expect(
      appTree.read('libs/mylib/vite.config.ts', 'utf-8')
    ).toMatchSnapshot();
    expect(getFormattedListChanges(appTree)).toMatchSnapshot();
    const expectedDeps = [
      '@builder.io/qwik-react',
      '@emotion/react',
      '@emotion/styled',
      '@types/react',
      '@types/react-dom',
      'react',
      'react-dom',
      '@mui/material',
      '@mui/x-data-grid',
    ];
    const deps = Object.keys(
      JSON.parse(appTree.read('package.json', 'utf-8')!).devDependencies
    );
    expect(expectedDeps.every((d) => deps.includes(d))).toBeTruthy();
  });

  it('should generate demo component without MUI dependencies', async () => {
    await reactLibraryGenerator(appTree, {
      ...options,
      installMUIExample: false,
    });
    const config = readProjectConfiguration(appTree, 'mylib');

    expect(config.targets!['build']).not.toBeDefined();
    expect(config.targets!['test']).toBeDefined();
    expect(config.targets!['lint']).toBeDefined();

    expect(config).toMatchSnapshot();
    expect(
      appTree.read('libs/mylib/vite.config.ts', 'utf-8')
    ).toMatchSnapshot();
    expect(getFormattedListChanges(appTree)).toMatchSnapshot();
    const expectedDeps = [
      '@builder.io/qwik-react',
      '@emotion/react',
      '@emotion/styled',
      '@types/react',
      '@types/react-dom',
      'react',
      'react-dom',
    ];
    const muiDeps = ['@mui/material', '@mui/x-data-grid'];
    const deps = Object.keys(
      JSON.parse(appTree.read('package.json', 'utf-8')!).devDependencies
    );
    expect(expectedDeps.every((d) => deps.includes(d))).toBeTruthy();
    expect(muiDeps.some((d) => deps.includes(d))).toBeFalsy();
  });

  it('should update app config if provided', async () => {
    await appGenerator(appTree, { name: 'myapp1' });
    await appGenerator(appTree, { name: 'myapp2' });
    await reactLibraryGenerator(appTree, {
      ...options,
      targetApps: 'myapp1, myapp2',
    });

    expect(appTree.read('apps/myapp1/vite.config.ts', 'utf-8')).toMatchSnapshot(
      'myapp1'
    );
    expect(appTree.read('apps/myapp2/vite.config.ts', 'utf-8')).toMatchSnapshot(
      'myapp2'
    );
  });
});
