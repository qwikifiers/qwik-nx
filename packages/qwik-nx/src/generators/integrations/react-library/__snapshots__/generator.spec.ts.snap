// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integrations/react-library generator should generate demo component without MUI dependencies 1`] = `
Object {
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "name": "mylib",
  "projectType": "library",
  "root": "libs/mylib",
  "sourceRoot": "libs/mylib/src",
  "tags": Array [],
  "targets": Object {
    "lint": Object {
      "executor": "@nrwl/linter:eslint",
      "options": Object {
        "lintFilePatterns": Array [
          "libs/mylib/**/*.{ts,tsx,js,jsx}",
        ],
      },
      "outputs": Array [
        "{options.outputFile}",
      ],
    },
    "test": Object {
      "executor": "@nrwl/vite:test",
      "options": Object {
        "passWithNoTests": true,
        "reportsDirectory": "../../coverage/libs/mylib",
      },
      "outputs": Array [
        "../..//coverage/libs/mylib",
      ],
    },
  },
}
`;

exports[`integrations/react-library generator should generate demo component without MUI dependencies 2`] = `
"import { qwikVite } from '@builder.io/qwik/optimizer';
import tsconfigPaths from 'vite-tsconfig-paths';
import { defineConfig } from 'vite';
import { qwikNxVite } from 'qwik-nx/plugins';
import { qwikReact } from '@builder.io/qwik-react/vite';
export default defineConfig({
  cacheDir: '../../node_modules/.vite/libs/mylib',
  plugins: [
    qwikNxVite(),
    qwikVite(),
    tsconfigPaths({ root: '../../' }),
    qwikReact(),
  ],
  test: {
    globals: true,
    cache: {
      dir: '../../node_modules/.vitest',
    },
    environment: 'node',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
    coverage: {
      reportsDirectory: '../../coverage/libs/mylib',
    },
  },
});
"
`;

exports[`integrations/react-library generator should generate demo component without MUI dependencies 3`] = `
Array [
  Object {
    "path": ".eslintignore",
    "type": "CREATE",
  },
  Object {
    "path": ".eslintrc.json",
    "type": "CREATE",
  },
  Object {
    "path": ".prettierignore",
    "type": "CREATE",
  },
  Object {
    "path": ".prettierrc",
    "type": "CREATE",
  },
  Object {
    "path": "apps/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "libs/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/.eslintrc.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/project.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/README.md",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/src/index.ts",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/src/lib/integration/demo.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/src/root.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/tsconfig.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/tsconfig.lib.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/tsconfig.spec.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/vite.config.ts",
    "type": "CREATE",
  },
  Object {
    "path": "nx.json",
    "type": "CREATE",
  },
  Object {
    "path": "package.json",
    "type": "CREATE",
  },
  Object {
    "path": "tsconfig.base.json",
    "type": "CREATE",
  },
]
`;

exports[`integrations/react-library generator should run successfully 1`] = `
Object {
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "name": "mylib",
  "projectType": "library",
  "root": "libs/mylib",
  "sourceRoot": "libs/mylib/src",
  "tags": Array [],
  "targets": Object {
    "lint": Object {
      "executor": "@nrwl/linter:eslint",
      "options": Object {
        "lintFilePatterns": Array [
          "libs/mylib/**/*.{ts,tsx,js,jsx}",
        ],
      },
      "outputs": Array [
        "{options.outputFile}",
      ],
    },
    "test": Object {
      "executor": "@nrwl/vite:test",
      "options": Object {
        "passWithNoTests": true,
        "reportsDirectory": "../../coverage/libs/mylib",
      },
      "outputs": Array [
        "../..//coverage/libs/mylib",
      ],
    },
  },
}
`;

exports[`integrations/react-library generator should run successfully 2`] = `
"import { qwikVite } from '@builder.io/qwik/optimizer';
import tsconfigPaths from 'vite-tsconfig-paths';
import { defineConfig } from 'vite';
import { qwikNxVite } from 'qwik-nx/plugins';
import { qwikReact } from '@builder.io/qwik-react/vite';
export default defineConfig({
  cacheDir: '../../node_modules/.vite/libs/mylib',
  plugins: [
    qwikNxVite(),
    qwikVite(),
    tsconfigPaths({ root: '../../' }),
    qwikReact(),
  ],
  test: {
    globals: true,
    cache: {
      dir: '../../node_modules/.vitest',
    },
    environment: 'node',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
    coverage: {
      reportsDirectory: '../../coverage/libs/mylib',
    },
  },
});
"
`;

exports[`integrations/react-library generator should run successfully 3`] = `
Array [
  Object {
    "path": ".eslintignore",
    "type": "CREATE",
  },
  Object {
    "path": ".eslintrc.json",
    "type": "CREATE",
  },
  Object {
    "path": ".prettierignore",
    "type": "CREATE",
  },
  Object {
    "path": ".prettierrc",
    "type": "CREATE",
  },
  Object {
    "path": "apps/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "libs/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/.eslintrc.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/project.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/README.md",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/src/index.ts",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/src/lib/integration/mui.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/src/root.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/tsconfig.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/tsconfig.lib.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/tsconfig.spec.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/mylib/vite.config.ts",
    "type": "CREATE",
  },
  Object {
    "path": "nx.json",
    "type": "CREATE",
  },
  Object {
    "path": "package.json",
    "type": "CREATE",
  },
  Object {
    "path": "tsconfig.base.json",
    "type": "CREATE",
  },
]
`;

exports[`integrations/react-library generator should update app config if provided: myapp1 1`] = `
"import { qwikVite } from '@builder.io/qwik/optimizer';
import { qwikCity } from '@builder.io/qwik-city/vite';
import { defineConfig } from 'vite';
import tsconfigPaths from 'vite-tsconfig-paths';
import { qwikNxVite } from 'qwik-nx/plugins';
import { qwikReact } from '@builder.io/qwik-react/vite';
export default defineConfig({
  cacheDir: '../../node_modules/.vite/apps/myapp1',
  plugins: [
    qwikNxVite(),
    qwikCity(),
    qwikVite({
      client: {
        outDir: '../../dist/apps/myapp1/client',
      },
      ssr: {
        outDir: '../../dist/apps/myapp1/server',
      },
    }),
    tsconfigPaths({ root: '../../' }),
    qwikReact(),
  ],
  server: {
    fs: {
      // Allow serving files from the project root
      allow: ['../../'],
    },
  },
  preview: {
    headers: {
      'Cache-Control': 'public, max-age=600',
    },
  },
  test: {
    globals: true,
    cache: {
      dir: '../../node_modules/.vitest',
    },
    environment: 'node',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
  },
});
"
`;

exports[`integrations/react-library generator should update app config if provided: myapp2 1`] = `
"import { qwikVite } from '@builder.io/qwik/optimizer';
import { qwikCity } from '@builder.io/qwik-city/vite';
import { defineConfig } from 'vite';
import tsconfigPaths from 'vite-tsconfig-paths';
import { qwikNxVite } from 'qwik-nx/plugins';
import { qwikReact } from '@builder.io/qwik-react/vite';
export default defineConfig({
  cacheDir: '../../node_modules/.vite/apps/myapp2',
  plugins: [
    qwikNxVite(),
    qwikCity(),
    qwikVite({
      client: {
        outDir: '../../dist/apps/myapp2/client',
      },
      ssr: {
        outDir: '../../dist/apps/myapp2/server',
      },
    }),
    tsconfigPaths({ root: '../../' }),
    qwikReact(),
  ],
  server: {
    fs: {
      // Allow serving files from the project root
      allow: ['../../'],
    },
  },
  preview: {
    headers: {
      'Cache-Control': 'public, max-age=600',
    },
  },
  test: {
    globals: true,
    cache: {
      dir: '../../node_modules/.vitest',
    },
    environment: 'node',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
  },
});
"
`;
